hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -1982,6 +1982,7 @@ void setPermission(String src, FsPermission permission) throws IOException {
     FileStatus auditStat;
     checkOperation(OperationCategory.WRITE);
     final FSPermissionChecker pc = getPermissionChecker();
+    FSPermissionChecker.setOperationType(operationName);
```

Comment:
There are other places that need to be patched with setOperationType After HDFS-7416 refactor, not all permission check is done in FSN.

Here is the list of missed ones:
FSDirSymlinkOp#createSymlinkInt()
NameNodeAdapter#getFileInfo()
NamenodeFsck#getBlockLocations()
FSNDNCache#addCacheDirective/removeCacheDirective/modifyCacheDirective/listCacheDirectives/listCachePools

Reply:
Thanks @xiaoyuyao for the review. 
* FSDirSymlinkOp#createSymlinkInt() is an exception. It doesn't check permission in the FSNamesystem so missed this one. Added.

* NameNodeAdapter#getFileInfo() is used only in tests. 
* NamenodeFsck#getBlockLocations() --> call it fsckGetBlockLocations to distinguish it from regular open operations.
* FSNDNCache#addCacheDirective/removeCacheDirective/modifyCacheDirective/listCacheDirectives/listCachePools --> done